ci-operator/step-registry/aws/provision/route53/private-hosted-zone/aws-provision-route53-private-hosted-zone-commands.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+REGION="${LEASED_RESOURCE}"
+CONFIG="${SHARED_DIR}/install-config.yaml"
+CLUSTER_NAME="$(/tmp/yq r "${CONFIG}" 'metadata.name')"
+
+route53_hosted_zone_name="${CLUSTER_NAME}.${BASE_DOMAIN}"
+VpcId=$(cat "${SHARED_DIR}/vpc_id")
+caller_reference_str=$route53_hosted_zone_name
+output_route53_json=${SHARED_DIR}/route53_hosted_private_zone.json
+
+echo -e "creating route53 hosted zone: ${route53_hosted_zone_name}"
+aws --region $REGION route53 create-hosted-zone --name ${route53_hosted_zone_name} --vpc VPCRegion=${REGION},VPCId=${VpcId} --caller-reference ${caller_reference_str} > ${output_route53_json}.creating
+
+hosted_zone_id=$(cat "${output_route53_json}.creating" | jq -r '.HostedZone.Id' | awk -F '/' '{printf $3}')
+change_id=$(cat "${output_route53_json}.creating" | jq -r '.ChangeInfo.Id' | awk -F '/' '{printf $3}')
+
+aws --region $REGION route53 wait resource-record-sets-changed --id ${change_id} &
+wait "$!"
+echo "Waited for hosted zone creation"
+
+aws --region $REGION route53 get-hosted-zone --id ${hosted_zone_id} > ${output_route53_json}
+# save hosted zone information to ${SHARED_DIR} for deprovision step
+echo "${hosted_zone_id}" > "${SHARED_DIR}/hosted_zone_id"
```

Comment:
I think it would be better to move this command after `hosted_zone_id=$(cat... `, as soon as we get the `hosted_zone_id`, we record it.

Reply:
Done